utils/managed_bytes.hh

```diff
@@ -394,6 +403,35 @@ public:
     void remove_prefix(size_t prefix_len);
 
     friend class managed_bytes_view;
+    friend class managed_bytes_view_fragment_iterator;
+};
+
+class managed_bytes_view_fragment_iterator : public managed_bytes_view_base {
+public:
+    managed_bytes_view_fragment_iterator() = default;
+    managed_bytes_view_fragment_iterator(const managed_bytes_view_fragment_iterator&) = default;
+    managed_bytes_view_fragment_iterator(const managed_bytes_view_base& x) noexcept
+            : managed_bytes_view_base(x)
+    {}
+
+    managed_bytes_view_fragment_iterator& operator=(const managed_bytes_view_fragment_iterator&) = default;
+
+    bytes_view operator*() const noexcept {
+        return _current_fragment;
+    }
+    const bytes_view* operator->() const noexcept {
+        return &_current_fragment;
+    }
+
+    managed_bytes_view_fragment_iterator& operator++() noexcept {
+        remove_prefix(_current_fragment.size());
```

Comment:
I think this is too heavy weight. This should be like 3 lines of code, no need to call the huge `remove_prefix()` to replace it.

Reply:
If it wasn't for `substr` I'd say you're right, but the way `substr` is defined, it may truncate the managed_bytes_view size so the iterator needs to take that into account and adjust the last fragment (dropping everything after it).

The reason `substr` is not actually truncating the view is that otherwise, it'll have to mess with `blob_storage`.
It can't modify the `blob_storage` it points to. They belong to the underlying `manage_bytes`.
So we'll need to provide it an interface to copy the last `blob_storage` fragment and modify its `frag_size` and `next` members.

Reply:
There is a simple solution to this: have a `_remains` member, from which you substract each time you move to the next fragment. When dereferencing you return `bytes_view(_current_fragment.data(), std::min(_remains, _current_fragment.size())`.

Reply:
In case the last fragment should be truncated (because of `substr()`) this will automatically return a `bytes_view()` with just the required part of it.

Reply:
Sounds good, I'll try that.